spec/js/post_office_search_spec.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { getByLabelText, getByRole, waitFor } from '@testing-library/dom';
+import '@testing-library/jest-dom';
+import { act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+const wait = (ms: number = 0) =>
+  new Promise<void>((res) => {
+    setTimeout(() => {
+      res();
+    }, ms);
+  });
+
+describe('Post Office Search', () => {
+  // This is simulated since the test env doesn't support fetch
+  const testServerPort = 9835;
+  const addressSearchUrl = `http://127.0.0.1:${testServerPort}/api/address_search`;
+  const locationsSearchUrl = `http://127.0.0.1:${testServerPort}/api/usps_locations`;
+
+  beforeEach(() => {
+    if ('fetch' in global) {
+      jest.spyOn(global, 'fetch');
+    } else {
+      // @ts-ignore
+      global.fetch = jest.fn();
+    }
+    (global.fetch as jest.Mock).mockImplementation(() => ({
+      get ok() {
+        return true;
+      },
+      json: jest.fn(() => []),
+      get headers() {
+        return new Headers();
+      },
+    }));
+  });
+
+  let container: HTMLDivElement;
+  let user: ReturnType<typeof userEvent['setup']>;
+  beforeEach(async () => {
+    user = userEvent.setup();
+    await act(async () => {
+      container = document.createElement('div');
+      container.id = 'post-office-search';
+      container.dataset.addressSearchUrl = addressSearchUrl;
+      container.dataset.locationsSearchUrl = locationsSearchUrl;
+      document.body.appendChild(container);
+      jest.isolateModules(() => {
+        // eslint-disable-next-line global-require
+        require('../../assets/js/post_office_search');
+      });
+      await wait();
```

Comment:
What does this wait for? Would it be more reliable to use DOM Testing Library's `waitFor` to test for something specific in the DOM if there's asynchronous initialization, rather than rely on time?

Reply:
This allows asynchronous actions taken by React to complete before the `act` function is finished, which prevents React from showing warning messages in this test.

Otherwise we get this error:
```
Warning: An update to AddressInput inside a test was not wrapped in act(...).
```

Reply:
Also, I want to leave assertions to the individual tests where possible instead of the before step.

Reply:
I more meant use `waitFor` as a way to check that the page is settled / ready for testing, not as an assertion in itself.